cluster/src/main/java/com/hartwig/pipeline/calling/structural/gridss/stage/Annotation.java

```diff
@@ -3,44 +3,49 @@
-    private final CommandFactory commandFactory;
-
-    @Value.Immutable
-    public interface AnnotationResult {
-        String annotatedVcf();
-
-        List<BashCommand> commands();
-    }
-
-    public Annotation(final CommandFactory commandFactory) {
-        this.commandFactory = commandFactory;
+import com.hartwig.pipeline.execution.vm.BashStartupScript;
+import com.hartwig.pipeline.execution.vm.OutputFile;
+import com.hartwig.pipeline.execution.vm.VmDirectories;
+
+public class Annotation extends SubStage {
+    private final String sampleBam;
+    private final String tumorBam;
+    private final String assemblyBam;
+    private final String rawVcf;
+    private final String referenceGenome;
+    private final String jointName;
+    private final String configFile;
+    private final String blacklist;
+
+    public Annotation(final String sampleBam, final String tumorBam, final String assemblyBam, final String rawVcf,
+                      final String referenceGenome, final String jointName, final String configFile, final String blacklist) {
+        super("annotation", OutputFile.GZIPPED_VCF);
+        this.sampleBam = sampleBam;
+        this.tumorBam = tumorBam;
+        this.assemblyBam = assemblyBam;
+        this.rawVcf = rawVcf;
+        this.referenceGenome = referenceGenome;
+        this.jointName = jointName;
+        this.configFile = configFile;
+        this.blacklist = blacklist;
     }
 
-    public AnnotationResult initialise(final String sampleBam, final String tumorBam, final String assemblyBam, final String rawVcf,
-            final String referenceGenome, final String tumorSampleName, final String configFile, final String blacklist) {
-        AnnotateVariants variants =
-                commandFactory.buildAnnotateVariants(sampleBam, tumorBam, assemblyBam, rawVcf, referenceGenome, tumorSampleName,
+    @Override
+    public BashStartupScript bash(OutputFile input, OutputFile output, BashStartupScript bash) {
+        String annotatedVcf = VmDirectories.outputFile(jointName + ".annotated_variants.vcf");
+        AnnotateVariants variants = new AnnotateVariants(sampleBam, tumorBam, assemblyBam, input.path(), referenceGenome, annotatedVcf,
                         configFile, blacklist);
-        AnnotateUntemplatedSequence untemplated =
-                commandFactory.buildAnnotateUntemplatedSequence(variants.resultantVcf(), referenceGenome, tumorSampleName);
-        BgzipCommand bgzip = commandFactory.buildBgzipCommand(untemplated.resultantVcf());
-        String finalOutputPath = format("%s.gz", untemplated.resultantVcf());
-        TabixCommand tabix = commandFactory.buildTabixCommand(finalOutputPath);
-
-        return ImmutableAnnotationResult.builder()
-                .annotatedVcf(finalOutputPath)
-                .commands(asList(variants, untemplated, bgzip, tabix))
-                .build();
+        String untemplatedOutputVcf = VmDirectories.outputFile(jointName + "_untemplated.vcf");
```

Comment:
Could inline some of this I think (eg bash.addCommand(new AnnotateUntemplatedSequence()).addCommand(new BgZipCommand). Might make it read slightly easier.